modules/dnn/src/cuda4dnn/csl/tensor.hpp

```diff
@@ -28,7 +28,7 @@
 #include <utility>
 
 #ifndef CSL_MAX_TENSOR_RANK
-    #define CSL_MAX_TENSOR_RANK 6
+    #define CSL_MAX_TENSOR_RANK 7
```

Comment:
Not sure why there is a rank limit. Could it be removed?

Reply:
I took a look at it. It seems that currently this variables is deeply rooted in CUDA backend. We can remove it from assertions and nothing happens. But to remove it completed quite a number of classes and function need to be refactored.

Reply:
Ok, we keep it then.